lib/modules/console_listener/index.js

```diff
@@ -0,0 +1,79 @@
+const utils = require('../../utils/utils.js');
+
+class ConsoleListener {
+    constructor(embark, options) {
+        this.logger = embark.logger;
+        this.ipc = options.ipc;  
+        this.addressToContract = [];
+        this.contractsConfig = embark.config.contractsConfig;
+        this.listenForLogRequests();
+    }
+
+    _updateContractList(){
+        Object.keys(this.contractsConfig.contracts).forEach(contractName => {
+            let contract = this.contractsConfig.contracts[contractName];
+            let address = contract.deployedAddress.toLowerCase();
+
+            if(!this.addressToContract[address]){
+                let funcSignatures = {};
+                contract.abiDefinition
+                    .filter(func => func.type == "function")
+                    .map(func => {
+                                return {
+                                    name: func.name + 
+                                            '(' + 
+                                            (func.inputs ? func.inputs.map(input => input.type).join(',') : '') +
+                                            ')',
+                                    abi: func,
+                                    functionName: func.name
+                                };
+                        })
+                    .forEach(func => {
+                        funcSignatures[utils.sha3(func.name).substring(0, 10)] = func;
```

Comment:
Can't you just put that in the previous map and make the map a forEach?
```
const name = func.name + 
   '(' + 
  (func.inputs ? func.inputs.map(input => input.type).join(',') : '') +
  ')';
funcSignatures[utils.sha3(name).substring(0, 10)] = {
  name,
  abi: func,
  functionName: func.name
};
```
Having less loops makes me happy